k8s/cmd/main.go

```diff
@@ -92,3 +103,63 @@ func tryDumpUsage() {
 		os.Exit(0)
 	}
 }
+func setupGenDoc() {
+	var gendoc = false
+	flag.BoolVar(&gendoc, "gendoc", false,
+		"Auto generate the documentation for the command line tool."+
+			" The generated documentation will be written to the"+
+			` "references.md" file under the current directory.`)
+	flag.Parse()
+	if gendoc {
+		if err := genDoc(cmd, "references.md"); err != nil {
+			util.ErrLogger.Fatalf("Failed to generate the documentation: %+v", err)
+		}
+	}
```

Comment:
Looks quite wired... Parsing twice?

Reply:
Yeah, I don't find a good idea to gen the doc unless we run the function manually.

Reply:
I have take a try.

Reply:
pushed.